app/src/main/java/com/io/movio/fragments/MovieListFragment.kt

```diff
@@ -0,0 +1,75 @@
+package com.io.movio.fragments
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.io.movio.R
+import com.io.movio.adaptor.Adaptor
+import com.io.movio.databinding.FragmentMovieListBinding
+import com.io.movio.models.Movie
+
+
+
+class MovieListFragment : Fragment(), Adaptor.ItemOnClickListener {
+    private var movies: Array<Movie> = arrayOf()
+    private lateinit var binding: FragmentMovieListBinding
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentMovieListBinding.inflate(inflater)
+
+        createMovieArray()
+        initRecyclerView()
+
+        return binding.root
+    }
+
+
+    override fun onItemClick(movie: Movie) {
+
+        val fragment = MovieDetailFragment.newInstance(movie.title, movie.description , movie.cast)
```

Comment:
Why don't you just pass the whole movie object, if in the future it has 10 properties are you going to pass them one by one?